### ISSUE TITLE
Improve Performance

### ISSUE THREAD

**Issue by slazyk** [FuzzyAutocomplete/FuzzyAutocompletePlugin#99]
Surprisingly, the plugin becomes slower on newer versions of Xcode. Need to re-evaluate performance bottlenecks. Last big performance optimisation was done when Xcode 5 was still around :)

Initial investigation of performance under Xcode 7.1 shows that getting a priority factor has (only now, or had it always?) a synchronized part, and can make parallel processing slower than single thread when many items pass the filtering.

Also, it takes surprisingly long (even given it does synchronize) and the priorities do not change when narrowing down.

- [x] use single thread for getting priority factors
- [x] cache priority factors for narrowing down
- [ ] make filtering before populating the cache also parallel
- [ ] investigate how Xcode filtering and scoring changed
- [ ] identify other bottlenecks

**Comment by lovelock** [comment]
Yeah, I run into this problem, too. Mac runs quite slow and hot when I swithed the plugin on, on which I run Xcode 7.1

**Comment by akring** [comment]
This is the problem man,it's much slower when I'm working on Swift.Wish we can get this update soon：）
